plonk/src/circuit/mod.rs

```diff
@@ -150,6 +151,46 @@ pub trait Circuit<F: Field> {
     /// Return error if the input variables are invalid.
     fn equal_gate(&mut self, a: Variable, b: Variable) -> Result<(), PlonkError>;
 
+    /// Constrain that `a` < `b`.
+    fn enforce_le(&mut self, a: Variable, b: Variable) -> Result<(), PlonkError>
+    where
+        F: PrimeField;
+
+    /// Constrain that`a` <= `b`.
+    fn enforce_leq(&mut self, a: Variable, b: Variable) -> Result<(), PlonkError>
+    where
+        F: PrimeField;
+
+    /// Constrain that `a` > `b`.
+    fn enforce_ge(&mut self, a: Variable, b: Variable) -> Result<(), PlonkError>
+    where
+        F: PrimeField;
+
+    /// Constrain that `a` >= `b`.
+    fn enforce_geq(&mut self, a: Variable, b: Variable) -> Result<(), PlonkError>
+    where
+        F: PrimeField;
```

Comment:
1. do you think this granular set of API is better than a single `enforce_cmp(a: Variable, b: Variable, ordering: Ordering, enforce_equality: bol)`? (I tend to prefer the compact single API)
2. even if we were to keep these, we should provide default implementation for everything except `enforce_le`, because the rest can be build just using available gate of `trait Circuit`?

Reply:
1. I think the current API set is more clear. Additionally, when you are using it, it's more convenient to write `enforce_le(a, b)` rather than `enforce_cmp(a, b, Ordering::Less, false)` and the code will be more understandable.
2. That's a good idea. Will do.